### ISSUE TITLE
Fluent Inheritance Mapping loses complex property of Base type during insert and update

### ISSUE THREAD

**Issue by dev-anton-ko** [linq2db/linq2db#3891]
### Steps to reproduce

There are 2 sets of identical classes hierarchy.
Mapping of the first one was configured using attributes and the second one using fluent mapping builder:
```c#
public class Name
{
  public string? First  { get; set; }
  public string? Second { get; set; }
}

[Table("Base")]
[InheritanceMapping(Code = 1, Type = typeof(ChildBase))]
[Column(MemberName ="Name.First", Name = "Name_First")]
[Column(MemberName = "Name.Second", Name = "Name_Second")]
public abstract class Base
{
  [Column, PrimaryKey]
  public int Id { get; set; }
  
  public Name? Name { get; set; }
  
  [Column(IsDiscriminator = true)]
  public int Type { get; set; }
}

[Table("Base")]
public class ChildBase : Base
{
  public ChildBase()
  {
    Type= 1;
  }
  
  [Column(Name = "Test_ChildId")]
  public int ChildId { get; set; }
}

public abstract class Base2
{
  public int Id { get; set; }
  
  public Name? Name { get; set; }
  
  public int Type { get; set; }
}

public class ChildBase2 : Base2
{
  public ChildBase2()
  {
  Type = 1;
  }
  
  public int ChildId { get; set; }
}
```

Test cases:
```c#
private MappingSchema CreateSchema()
{
	var cnt = TestUtils.GetNext().ToString();
	var ms = new MappingSchema(cnt);

	var mappingBuilder = ms.GetFluentMappingBuilder();

	mappingBuilder.Entity<Base2>()
		.HasTableName("Base2")
		.Inheritance(x => x.Type, 1, typeof(ChildBase2))
		.Property(x => x.Id).IsPrimaryKey()
		.Property(x => x.Name!.First).HasColumnName("Name_First")
		.Property(x => x.Name!.Second).HasColumnName("Name_Second");

	mappingBuilder.Entity<ChildBase2>()
		.HasTableName("Base2")
		.Property(x => x.ChildId).HasColumnName("Test_ChildId");

	return ms;
}

[Test]
public void Test1([InsertOrUpdateDataSources] string context)
{
	using (new DisableBaseline("TODO: debug reason for inconsistent column order"))
	using (var db = GetDataContext(context))
	using (db.CreateLocalTable<Base>())
	{
		var child = new ChildBase()
		{
			Id= 1,
			ChildId= 2,
			Name = new Name()
			{
				First = "First",
				Second = "Second"
			}
		};

		db.Insert(child);
                /* Insert Query:

                    INSERT INTO [Base]
                    (
	                    [Type],
	                    [Id],
	                    [Test_ChildId],
	                    [Name_First],
	                    [Name_Second]
                    )
                    VALUES
                    (
	                    @Type_1,
	                    @Id,
	                    @ChildId,
	                    @Name_First,
	                    @Name_Second
                    )
                */
		child.Name.First = "First1";
		db.Update(child);
                
                /* Update Query:
                  UPDATE
	                  [Base]
                  SET
	                  [Type] = @Type_1,
	                  [Test_ChildId] = @ChildId,
	                  [Name_First] = @Name_First,
	                  [Name_Second] = @Name_Second
                  WHERE
	                  [Base].[Id] = @Id
                */

	}

}

[Test]
public void Test2([InsertOrUpdateDataSources] string context)
{
	var ms = CreateSchema();

	using (new DisableBaseline("TODO: debug reason for inconsistent column order"))
	using (var db = GetDataContext(context, ms))
	using (db.CreateLocalTable<Base2>())
	{
		var child = new ChildBase2()
		{
			Id= 1,
			ChildId= 2,
			Name = new Name()
			{
				First = "First",
				Second = "Second"
			}
		};

		db.Insert(child);
                /* Insert Query:
                
                INSERT INTO [Base2]
                (
	                [Type],
	                [Id],
	                [Test_ChildId]
                )
                VALUES
                (
	                @Type_1,
	                @Id,
	                @ChildId
                )
                */
		
                child.Name.First = "First1";
		db.Update(child);
                
                /* Update Query:
                UPDATE
	                [Base2]
                SET
	                [Type] = @Type_1,
	                [Test_ChildId] = @ChildId
                WHERE
	                [Base2].[Id] = @Id
                */
	}
}
```
Complex property *Name* is missing in the queries, created by *Test2*

Note:
If I exclude configuring of the ChildBase2 type - queries will be identical (for examples, if changing of the column name is not required):

```c#
private MappingSchema CreateSchema()
{
	var cnt = TestUtils.GetNext().ToString();
	var ms = new MappingSchema(cnt);

	var mappingBuilder = ms.GetFluentMappingBuilder();

	mappingBuilder.Entity<Base2>()
		.HasTableName("Base2")
		.Inheritance(x => x.Type, 1, typeof(ChildBase2))
		.Property(x => x.Id)
		.Property(x => x.Name!.First).HasColumnName("Name_First")
		.Property(x => x.Name!.Second).HasColumnName("Name_Second");

	// mappingBuilder.Entity<ChildBase2>()
	//	.HasTableName("Base2")
	//	.Property(x => x.ChildId).HasColumnName("Test_ChildId");

	return ms;
}
```
### Environment details

`Linq To DB` version: 4.4.0

Database (with version): MS SQL 2019. Reproduced on the tests

Operating system: *Windows 11*

.NET Version: *.Net Core 6, .Net Core 7*

**Comment by dev-anton-ko** [comment]
Additional: If I remove Table Name from ChildBase2 declaration:
```c#
	mappingBuilder.Entity<ChildBase2>()
	    //.HasTableName("Base2")
	    .Property(x => x.ChildId).HasColumnName("Test_ChildId");
```
Insert & Update queries would be created properly until I add another complex property to the ChildBase2 declaration.
In such case Insert & Update will use Table Name = "ChildBase2":

_Insert into ChildBase2...._ or _Update ChildBase2...._

**Comment by MaceWindu** [comment]
It is not clear what you try to achieve. Multi-table inheritance not supported currently.

If you just want to use base type properties, it is not clear why you add inheritance attributes instead of base class.

**Comment by dev-anton-ko** [comment]
> It is not clear what you try to achieve. Multi-table inheritance not supported currently.
> 
> If you just want to use base type properties, it is not clear why you add inheritance attributes instead of base class.
> 
> Also multiple classes with inheritance mappings not supported in inheritance hierarchy.
> 
> I'm closing it as we already have issues tracking multi-table inheritance and/or multiple types with inheritance mapping in hierarchy.
> 
> If you have different case, please add more details what kind of model you try to setup

There is no multi-table inheritance. 
Main problem: 
* Configuring through the attributes is working properly
* Configuring through the fluent api - leads to the wrong queries creation

I will check on the 6.* and add details then.

**Comment by MaceWindu** [comment]
My bad. Retested it again. This was fixed some time ago starting from `5.0.0-preview.2` release. I see you use 4.4.0
